### ISSUE TITLE
Provide alert in UI when a rule is running for too long

### ISSUE THREAD

**Issue by asachs01** [Graylog2/graylog2-server#10259]
## What?

As an end user, I'd like to be warned if a pipeline rule is taking longer to run than I expect it too. 

## Why?

When processing slows down, or a user may have inadvertently created a rule that is not scoped well, this would help provide a warning and surface the issue more quickly. See HS-318695282

**Comment by l0calhost** [comment]
Thats what "Debug Metrics" are for, right?
![image](https://user-images.githubusercontent.com/8111901/111743913-a92b8c80-888a-11eb-8c03-0bb61c809011.png)

**Comment by asachs01** [comment]
@l0calhost that helps insomuch as it's a tool during the troubleshooting process, but we don't alert or give users any indication that a pipeline rule might be going off the rails. The idea here is to surface any rule issues more or less immediately and shorten the amount of time folks spend troubleshooting _before_ they find out that it's a pipleline rule that's the issue.

**Comment by l0calhost** [comment]
@asachs01 Fair enough. I'm looking at this from a corporate perspective and from there, a publication of metrics to be examined out of graylog is the way to visualize and assess the data.

Is there an idea about how to set "longer to run than I expect it to" limits?

**Comment by asachs01** [comment]
@l0calhost right--ideally, Graylog users would be capturing, storing and visualizing Graylog metrics. Those metrics, aside from being exposed in the UI would also be surfaced via a metrics reporter plugin IIRC. The challenge as I'm modeling it, is that there's a seemingly wide gap in technical maturity in a lot of orgs. There are folks who have mature observability practices in place, and those who haven't even really done the bare minimum in terms of ensuring that they have insight as to what's going on in their systems. 

Though as I'm reading this:
> Is there an idea about how to set "longer to run than I expect it to" limits?

What I'm reading is: "How would I benchmark a pipeline rule to know if it is actually running too long?" There's the simulator, which already provides insight as to the time it takes to process a message through a pipeline. The challenge is that the simulator, as far as I understand it, just looks at how long it took for a pipeline to process a message, now how long a particular rule within a pipeline took. So that could be a whole different can of worms, because I think we'd need to enable users to _easily_ benchmark the rules, or gather metrics for each rule so that they would have an idea of what to set for a threshold when creating the alert.
